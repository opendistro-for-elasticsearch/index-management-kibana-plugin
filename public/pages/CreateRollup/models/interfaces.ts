/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { ManagedCatIndex } from "../../../../server/models/interfaces";
import { FieldItem } from "../../../../models/interfaces";

export interface MappingsItem {
  properties?: {};
  type?: string;
}

export interface DimensionItem {
  sequence: number;
  field: FieldItem;
  aggregationMethod: string;
  interval?: number;
}

export interface MetricItem {
  source_field: FieldItem;
  all: boolean;
  min: boolean;
  max: boolean;
  sum: boolean;
  avg: boolean;
  value_count: boolean;
}

//From backend
export interface RollupDimensionItem {
  date_histogram?: {
    fixed_interval?: string;
    calendar_interval?: string;
    source_field: string;
    target_field: string;
    timezone: string;
  };
  terms?: {
    source_field: string;
    target_field: string;
  };
  histogram?: {
    source_field: string;
    target_field: string;
    interval: number;
  };
}

export interface RollupMetricItem {
  source_field: string;
  metrics: [
    {
      min?: Object;
      max?: Object;
      sum?: Object;
      avg?: Object;
      value_count?: Object;
    }
  ];
}
