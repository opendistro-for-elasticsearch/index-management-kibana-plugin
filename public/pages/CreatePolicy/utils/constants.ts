/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "A simple default policy that changes the replica count between hot and cold states.",
      default_state: "hot",
      states: [
        {
          name: "hot",
          actions: [{ replica_count: { number_of_replicas: 5 } }],
          transitions: [{ state_name: "cold", conditions: { min_index_age: "30d" } }],
        },
        {
          name: "cold",
          actions: [{ replica_count: { number_of_replicas: 2 } }],
          transitions: [],
        },
      ],
    },
  },
  null,
  4
);
