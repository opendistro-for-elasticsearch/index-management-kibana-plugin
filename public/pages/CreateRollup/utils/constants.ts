/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

//Added last_updated_time and start_time to avoid error, but they should be system generated.
//Added a delay as 0, but this should be optional, so null should be fine.
export const EMPTY_ROLLUP = JSON.stringify({
  rollup: {
    continuous: false,
    description: "",
    dimensions: [
      {
        date_histogram: {
          source_field: "",
          fixed_interval: "2ms",
          timezone: "Africa/Abidjan",
        },
      },
    ],
    enabled: false,
    metrics: [],
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];
