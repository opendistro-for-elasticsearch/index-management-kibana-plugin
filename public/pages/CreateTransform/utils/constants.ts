/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const EMPTY_TRANSFORM = JSON.stringify({
  transform: {
    description: "",
    enabled: true,
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "m", text: "Minute" },
  { value: "h", text: "Hour" },
  { value: "d", text: "Day" },
  { value: "w", text: "Week" },
  { value: "M", text: "Month" },
  { value: "q", text: "Quarter" },
  { value: "y", text: "Year" },
];

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];

export const ExecutionFrequencyDefinitionOptions = [
  { value: "fixed", text: "Define by fixed interval" },
  { value: "cron", text: "Define by cron expression" },
];

export const DefaultSampleDataSize = 50;
