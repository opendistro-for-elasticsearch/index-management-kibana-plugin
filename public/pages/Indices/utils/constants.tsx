/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { EuiHealth, EuiTableFieldDataColumnType } from "@elastic/eui";
import { ManagedCatIndex } from "../../../../server/models/interfaces";
import { SortDirection } from "../../../utils/constants";

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  size: 20,
  search: "",
  sortField: "name",
  sortDirection: SortDirection.DESC,
};

const HEALTH_TO_COLOR: {
  [health: string]: string;
  green: string;
  yellow: string;
  red: string;
} = {
  green: "success",
  yellow: "warning",
  red: "danger",
};

export const indicesColumns: EuiTableFieldDataColumnType<ManagedCatIndex>[] = [
  {
    field: "index",
    name: "Index",
    sortable: true,
    truncateText: true,
    textOnly: true,
    width: "250px",
    render: (index: string) => <span title={index}>{index}</span>,
  },
  {
    field: "health",
    name: "Health",
    sortable: true,
    truncateText: true,
    textOnly: true,
    align: "right",
    render: (health: string, item: ManagedCatIndex) => {
      const color = health ? HEALTH_TO_COLOR[health] : "subdued";
      const text = health || item.status;
      return <EuiHealth color={color}>{text}</EuiHealth>;
    },
  },
  {
    field: "managed",
    name: "Managed by Policy",
    sortable: false,
    truncateText: true,
    textOnly: true,
    align: "right",
    width: "150px",
  },
  {
    field: "status",
    name: "Status",
    sortable: true,
    truncateText: true,
    textOnly: true,
    align: "right",
  },
  {
    field: "store.size",
    name: "Total size",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "pri.store.size",
    name: "Primaries size",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "docs.count",
    name: "Total documents",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
    render: (count: string) => <span title={count}>{count}</span>,
  },
  {
    field: "docs.deleted",
    name: "Deleted documents",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
    render: (deleted: string) => <span title={deleted}>{deleted}</span>,
  },
  {
    field: "pri",
    name: "Primaries",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "rep",
    name: "Replicas",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
];
