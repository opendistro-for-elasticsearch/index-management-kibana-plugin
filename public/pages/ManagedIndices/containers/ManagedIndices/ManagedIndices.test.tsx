/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import { HashRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent, wait } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { CoreStart } from "kibana/public";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import ManagedIndices from "./ManagedIndices";
import { TEXT } from "../../components/ManagedIndexEmptyPrompt/ManagedIndexEmptyPrompt";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import { ManagedIndexItem } from "../../../../../models/interfaces";
import { CoreServicesConsumer, CoreServicesContext } from "../../../../components/core_services";

function renderWithRouter(Component: React.ComponentType<any>) {
  return {
    ...render(
      <Router>
        <Switch>
          <Route
            path={ROUTES.MANAGED_INDICES}
            render={(props) => (
              <CoreServicesContext.Provider value={coreServicesMock}>
                <ServicesContext.Provider value={browserServicesMock}>
                  <ModalProvider>
                    <ServicesConsumer>{(services) => services && <ModalRoot services={services} />}</ServicesConsumer>
                    <CoreServicesConsumer>
                      {(core: CoreStart | null) => (
                        <ServicesConsumer>
                          {({ managedIndexService }: any) => <Component managedIndexService={managedIndexService} {...props} />}
                        </ServicesConsumer>
                      )}
                    </CoreServicesConsumer>
                  </ModalProvider>
                </ServicesContext.Provider>
              </CoreServicesContext.Provider>
            )}
          />
          <Redirect from="/" to={ROUTES.MANAGED_INDICES} />
        </Switch>
      </Router>
    ),
  };
}

describe("<ManagedIndices /> spec", () => {
  it("renders the component", async () => {
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { managedIndices: [], totalManagedIndices: 0 } });
    const { container } = renderWithRouter(ManagedIndices);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("shows LOADING on mount", async () => {
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { managedIndices: [], totalManagedIndices: 0 } });
    const { getByText } = renderWithRouter(ManagedIndices);

    getByText(TEXT.LOADING);
  });

  it("sets breadcrumbs when mounting", async () => {
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { managedIndices: [], totalManagedIndices: 0 } });
    renderWithRouter(ManagedIndices);

    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledWith([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.MANAGED_INDICES]);
  });

  it("loads managed indices", async () => {
    const managedIndices = [
      {
        index: "index_1",
        indexUuid: "index_1_uuid",
        policyId: "some_policy",
        policySeqNo: 1,
        policyPrimaryTerm: 1,
        policy: null,
        enabled: true,
        managedIndexMetaData: null,
      },
    ];
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { managedIndices, totalManagedIndices: 1 } });
    const { getByText } = renderWithRouter(ManagedIndices);
    await wait();

    await wait(() => getByText("index_1"));
  });

  it("adds error toaster when get managed indices has error", async () => {
    browserServicesMock.managedIndexService.getManagedIndices = jest.fn().mockResolvedValue({ ok: false, error: "some error" });
    renderWithRouter(ManagedIndices);

    await wait();

    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledWith("some error");
  });

  it("adds error toaster when get managed indices throws error", async () => {
    browserServicesMock.managedIndexService.getManagedIndices = jest.fn().mockRejectedValue(new Error("rejected error"));
    renderWithRouter(ManagedIndices);

    await wait();

    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledWith("rejected error");
  });

  it("can remove a policy from a managed index", async () => {
    const managedIndices = [
      {
        index: "index_1",
        indexUuid: "index_1_uuid",
        policyId: "some_policy",
        policySeqNo: 1,
        policyPrimaryTerm: 1,
        policy: null,
        enabled: true,
        managedIndexMetaData: null,
      },
    ];
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { managedIndices, totalManagedIndices: 1 } });
    browserServicesMock.managedIndexService.removePolicy = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { updatedIndices: 1, failures: false, failedIndices: [] } });
    const { getByText, getByTestId } = renderWithRouter(ManagedIndices);

    await wait(() => getByText("index_1"));

    expect(getByTestId("Remove policyButton")).toBeDisabled();

    userEvent.click(getByTestId("checkboxSelectRow-index_1"));

    expect(getByTestId("Remove policyButton")).toBeEnabled();

    userEvent.click(getByTestId("Remove policyButton"));
    await wait(() => getByTestId("confirmationModalActionButton"));
    userEvent.click(getByTestId("confirmationModalActionButton"));
    await wait();

    expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Removed policy from 1 managed indices");
  });

  it("sorts/paginates the table", async () => {
    const managedIndices = new Array(40).fill(null).map((_, index) => ({
      index: `index_${index}`,
      indexUuid: `index_${index}_uuid`,
      policyId: "some_policy",
      policySeqNo: 1,
      policyPrimaryTerm: 1,
      policy: null,
      enabled: true,
      managedIndexMetaData: null,
    }));
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValueOnce({ ok: true, response: { managedIndices: managedIndices.slice(0, 20), totalManagedIndices: 40 } })
      .mockResolvedValueOnce({ ok: true, response: { managedIndices: managedIndices.slice(20), totalManagedIndices: 40 } })
      .mockResolvedValueOnce({
        ok: true,
        response: {
          managedIndices: managedIndices.sort().slice(0, 20),
          totalManagedIndices: 40,
        },
      });

    const { getByText, getByTestId, getAllByTestId, queryByText } = renderWithRouter(ManagedIndices);

    // should load managed indices 0-19 on first load
    await wait(() => getByText("index_0"));
    expect(queryByText("index_39")).toBeNull();

    fireEvent.click(getAllByTestId("pagination-button-next")[0]);

    // should load managed indices 20-39 after clicking next
    await wait(() => getByText("index_39"));
    expect(queryByText("index_0")).toBeNull();

    // @ts-ignore
    fireEvent.click(getByTestId("tableHeaderCell_index_0").firstChild);

    // should load managed indices 0-19 after clicking sort (defaults to asc) on index
    await wait(() => getByText("index_0"));
    expect(queryByText("index_39")).toBeNull();
  });

  it("retry disables/enables itself when selecting a failed/non-failed managed indices", async () => {
    const managedIndices: ManagedIndexItem[] = [
      {
        index: "index_1",
        indexUuid: "index_1_uuid",
        policyId: "some_policy",
        policySeqNo: 1,
        policyPrimaryTerm: 1,
        policy: {
          description: "some description",
          default_state: "two",
          states: [
            { name: "two", actions: [{ delete: {} }], transitions: [] },
            { name: "three", actions: [{ delete: {} }], transitions: [] },
          ],
        },
        enabled: true,
        managedIndexMetaData: {
          index: "index_1",
          indexUuid: "index_1_uuid",
          policyId: "some_policy",
          policyPrimaryTerm: 1,
          policySeqNo: 1,
          state: { name: "two", startTime: Date.now() },
          retryInfo: { failed: false, consumedRetries: 0 },
        },
      },
      {
        index: "index_2",
        indexUuid: "index_2_uuid",
        policyId: "some_policy",
        policySeqNo: 1,
        policyPrimaryTerm: 1,
        policy: {
          description: "some description",
          default_state: "two",
          states: [
            { name: "two", actions: [{ delete: {} }], transitions: [] },
            { name: "three", actions: [{ delete: {} }], transitions: [] },
          ],
        },
        enabled: false,
        managedIndexMetaData: {
          index: "index_2",
          indexUuid: "index_2_uuid",
          policyId: "some_policy",
          policyPrimaryTerm: 1,
          policySeqNo: 1,
          state: { name: "two", startTime: Date.now() },
          retryInfo: { failed: true, consumedRetries: 0 },
        },
      },
    ];
    browserServicesMock.managedIndexService.getManagedIndices = jest
      .fn()
      .mockResolvedValueOnce({ ok: true, response: { managedIndices: managedIndices, totalManagedIndices: 2 } });

    const { getByText, getByTestId } = renderWithRouter(ManagedIndices);

    await wait(() => getByText("index_1"));

    expect(getByTestId("Retry policyButton")).toBeDisabled();

    userEvent.click(getByTestId("checkboxSelectRow-index_2"));

    expect(getByTestId("Retry policyButton")).toBeEnabled();

    userEvent.click(getByTestId("checkboxSelectRow-index_1"));

    expect(getByTestId("Retry policyButton")).toBeDisabled();
  });
});
