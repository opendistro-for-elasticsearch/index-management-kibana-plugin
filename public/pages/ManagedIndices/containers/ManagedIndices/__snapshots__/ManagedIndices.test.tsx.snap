// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManagedIndices /> spec renders the component 1`] = `
<div
  style="padding: 0px 25px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem"
    >
      <h1
        class="euiTitle euiTitle--large"
      >
        Managed Indices
      </h1>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <a
        class="euiButton euiButton--primary"
        data-test-subj="changePolicyButton"
        href="opendistro_index_management_kibana#/change-policy"
        rel="noreferrer"
      >
        <span
          class="euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Change policy
          </span>
        </span>
      </a>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--large"
        >
          Indices
        </h3>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <button
                  class="euiButton euiButton--primary"
                  data-test-subj="Edit rollover aliasButton"
                  disabled=""
                  type="button"
                >
                  <span
                    class="euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      Edit rollover alias
                    </span>
                  </span>
                </button>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <button
                  class="euiButton euiButton--primary"
                  data-test-subj="Remove policyButton"
                  disabled=""
                  type="button"
                >
                  <span
                    class="euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      Remove policy
                    </span>
                  </span>
                </button>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <button
                  class="euiButton euiButton--primary"
                  data-test-subj="Retry policyButton"
                  disabled=""
                  type="button"
                >
                  <span
                    class="euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      Retry policy
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        style="padding: 0px 5px;"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldSearch euiFieldSearch--fullWidth"
                placeholder="Search"
                type="search"
                value=""
              />
              <div
                class="euiFormControlLayoutIcons"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  EuiIconMock
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="max-width: 250px;"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper euiSuperDatePicker__flexWrapper--isAutoRefreshOnly"
          >
            <div
              class="euiFlexItem"
            >
              <div
                class="euiFormControlLayout euiFormControlLayout--group euiSuperDatePicker"
              >
                <div
                  class="euiPopover euiPopover--anchorDownLeft"
                  id="QuickSelectPopover"
                >
                  <div
                    class="euiPopover__anchor euiQuickSelectPopover__anchor"
                  >
                    <button
                      aria-label="Date quick select"
                      class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--iconRight euiFormControlLayout__prepend"
                      data-test-subj="superDatePickerToggleQuickMenuButton"
                      type="button"
                    >
                      <span
                        class="euiButtonEmpty__content"
                      >
                        EuiIconMock
                        <span
                          class="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                        >
                          EuiIconMock
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <div
                    class="euiDatePickerRange euiDatePickerRange--readOnly euiDatePickerRange--inGroup"
                  >
                    <span
                      class="euiSuperDatePicker__prettyFormat"
                    >
                      Off
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
      />
      <div
        class="euiBasicTable"
      >
        <div>
          <div
            class="euiTableHeaderMobile"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiCheckbox"
                >
                  <input
                    aria-label="Select all rows"
                    class="euiCheckbox__input"
                    disabled=""
                    id="_selection_column-checkbox_some_make_id"
                    type="checkbox"
                  />
                  <div
                    class="euiCheckbox__square"
                  />
                  <label
                    class="euiCheckbox__label"
                    for="_selection_column-checkbox_some_make_id"
                  >
                    Select all rows
                  </label>
                </div>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiTableSortMobile"
                >
                  <div
                    class="euiPopover euiPopover--anchorDownRight"
                  >
                    <div
                      class="euiPopover__anchor"
                    >
                      <button
                        class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--iconRight euiButtonEmpty--flushRight"
                        type="button"
                      >
                        <span
                          class="euiButtonEmpty__content"
                        >
                          EuiIconMock
                          <span
                            class="euiButtonEmpty__text"
                          >
                            Sorting
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <table
            class="euiTable euiTable--responsive"
          >
            <caption
              class="euiScreenReaderOnly euiTableCaption"
            />
            <thead>
              <tr>
                <th
                  class="euiTableHeaderCellCheckbox"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <div
                      class="euiCheckbox euiCheckbox--inList euiCheckbox--noLabel"
                    >
                      <input
                        aria-label="Select all rows"
                        class="euiCheckbox__input"
                        data-test-subj="checkboxSelectAll"
                        disabled=""
                        id="_selection_column-checkbox_some_make_id"
                        type="checkbox"
                      />
                      <div
                        class="euiCheckbox__square"
                      />
                    </div>
                  </div>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="descending"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_index_0"
                  role="columnheader"
                  scope="col"
                  style="width: 150px;"
                >
                  <button
                    class="euiTableHeaderButton euiTableHeaderButton-isSorted"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Index; Sorted in descending order"
                      >
                        Index
                      </span>
                      EuiIconMock
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="none"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_policyId_1"
                  role="columnheader"
                  scope="col"
                  style="width: 150px;"
                >
                  <button
                    class="euiTableHeaderButton"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Policy"
                      >
                        Policy
                      </span>
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_managedIndexMetaData.state.name_2"
                  role="columnheader"
                  scope="col"
                  style="width: 150px;"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="State"
                    >
                      State
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_managedIndexMetaData.action.name_3"
                  role="columnheader"
                  scope="col"
                  style="width: 150px;"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Action"
                    >
                      Action
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_managedIndexMetaData.info_4"
                  role="columnheader"
                  scope="col"
                  style="width: 150px;"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Info"
                    >
                      Info
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_index_5"
                  role="columnheader"
                  scope="col"
                  style="width: 150px;"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Job Status"
                    >
                      Job Status
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="euiTableRow"
              >
                <td
                  class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                  colspan="7"
                >
                  <div
                    class="euiTableCellContent euiTableCellContent--alignCenter"
                  >
                    <span
                      class="euiTableCellContent__text"
                    >
                      <div
                        class="euiEmptyPrompt"
                        style="max-width: 45em;"
                      >
                        <span
                          class="euiTextColor euiTextColor--subdued"
                        >
                          <div
                            class="euiText euiText--medium"
                          >
                            <div
                              class="euiText euiText--medium"
                            >
                              <p>
                                Loading managed indices...
                              </p>
                            </div>
                          </div>
                        </span>
                      </div>
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
